### PR TITLE
Chore/sof 6429

### PR DESCRIPTION
**Pull request by spectre007**
This PR concerns the addition of a second function to generate the `dependencies` block using `"oneOf"` instead of `"enum"`.

### CODE REVIEW

File: src/utils/schemas.js [hunk 1]
```diff
@@ -75,7 +88,38 @@ export function buildDependencies(nodes) {
...
+        },
+    };
+}
+
+/**
+ * @summary Recursively generate `dependencies` for RJSF schema based on tree.
+ * @param {Object[]} nodes - Array of nodes (e.g. `[tree]` or `node.children`)
+ * @returns {{}|{dependencies: {}}}
+ */
+export function buildDependenciesOneOf(nodes) {
```
Comment (timurbazhirov) [inline src/utils/schemas.js:104]: Should we move this to a file named `tree.js`?

File: tests/utils.schemas.tests.js [hunk 2]
```diff
@@ -161,6 +195,23 @@ describe("RJSF schema", () => {
...
     });
+
+    it("title and const (enum) can be added to schema properties", () => {
+        const rjsfSchema = getSchemaWithDependencies({
+            schema: DFT_SCHEMA,
+            nodes: [TREE],
+            modifyProperties: true,
+            useEnum: false,
+        });
+        // console.log(JSON.stringify(rjsfSchema, null, 4));
```
Comment (timurbazhirov) [inline tests/utils.schemas.tests.js:206]: Remove

File: tests/utils.schemas.tests.js [hunk 3]
```diff
@@ -1,57 +1,62 @@
...
 describe("RJSF schema", () => {
     const TREE = {
         path: "/dft",
-        dataSelector: { key: "type", value: "data.type.slug", name: "data.type.name" },
+        dataSelector: { key: "type", value: "data.type.enum[0]", name: "data.type.title" },
         data: {
             type: {
-                slug: "dft",
-                name: "Density Functional Theory",
+                enum: ["dft"],
```
Comment (timurbazhirov) [inline tests/utils.schemas.tests.js:16]: How do we decide what to use - enum or slug??

File: src/utils/schemas.js (hunk 1 above)
Comment (spectre007) [reply src/utils/schemas.js:100]: It makes use of a tree structure, yes, but the result of this function is really only useful in the context of schemas. In my opinion, it is already in a good place.

File: tests/utils.schemas.tests.js [hunk 4]
```diff
@@ -1,57 +1,62 @@
...
                         },
                     },
                     {
                         path: "/dft/lda/pz",
                         dataSelector: {
                             key: "functional",
-                            value: "data.functional.slug",
-                            name: "data.functional.name",
+                            value: "data.functional.enum[0]",
+                            name: "data.functional.title",
```
Comment (timurbazhirov) [inline tests/utils.schemas.tests.js:54]: Per our discussion:

1. Let's add type to know whether we are dealing with a string or object for enum[0] use
2. When object, we can have `value: data.functional`, `name: data.functional.name`

**Pull request closed without merging**